cpp/src/arrow/compute/row/grouper.cc

```diff
@@ -55,18 +55,15 @@ using group_id_t = std::remove_const<decltype(kNoGroupId)>::type;
 using GroupIdType = CTypeTraits<group_id_t>::ArrowType;
 auto g_group_id_type = std::make_shared<GroupIdType>();
 
-template <typename Value>
-ARROW_DEPRECATED("Deprecated in 18.0.0 along with GetSegments.")
-Status CheckForGetNextSegment(const std::vector<Value>& values, int64_t length,
-                              int64_t offset, const std::vector<TypeHolder>& key_types) {
-  DCHECK_GE(offset, 0);
-  DCHECK_LE(offset, length);
-  if (values.size() != key_types.size()) {
+Status CheckForGetSegments(const ExecSpan& batch,
+                           const std::vector<TypeHolder>& key_types) {
+  DCHECK_LE(0, batch.length);
```

Comment:
I guess we don't really need this checking?

Reply:
Yeah, true 😊 🤦‍♀️